projects/vdk-plugins/vdk-data-sources/src/vdk/plugin/data_sources/auto_generated.py

```diff
@@ -0,0 +1,107 @@
+# Copyright 2021-2023 VMware, Inc.
+# SPDX-License-Identifier: Apache-2.0
+from datetime import datetime
+from typing import Iterator
+from typing import List
+
+from vdk.plugin.data_sources.config import config_class
+from vdk.plugin.data_sources.config import config_field
+from vdk.plugin.data_sources.data_source import DataSourcePayload
+from vdk.plugin.data_sources.data_source import IDataSource
+from vdk.plugin.data_sources.data_source import (
+    IDataSourceConfiguration,
+)
+from vdk.plugin.data_sources.data_source import IDataSourceStream
+from vdk.plugin.data_sources.factory import data_source
+from vdk.plugin.data_sources.state import IDataSourceState
+
+
+@config_class(
+    name="auto_generated", description="Configuration for Auto generated data source"
+)
+class AutoGeneratedDataSourceConfiguration(IDataSourceConfiguration):
+    num_records: int = config_field(
+        description="Number of records to return", default=2
+    )
+    include_metadata: bool = config_field(
+        description="If true autogenerated metadata is included in the response",
+        default=False,
+    )
+    num_streams: int = config_field(
+        description="The number of streams the data source would have", default=1
+    )
+
+
+class AutoGeneratedDataSourceStream(IDataSourceStream):
+    """
+    A single data stream in AutoGeneratedDataSource
+    """
+
+    def name(self) -> str:
+        return f"stream_{self._stream_number}"
+
+    def __init__(
+        self,
+        config: AutoGeneratedDataSourceConfiguration,
+        stream_number: int,
+        start_id: int = 0,
+    ):
+        self._config = config
+        self._stream_number = stream_number
+        self._data = self._generate_test_data(start_id)
+
+    def _generate_test_data(self, start_id: int) -> List[DataSourcePayload]:
+        generated_data = []
+        for i in range(self._config.num_records):
+            data = {
```

Comment:
Hmm, yeah I probably could have come with some more interesting example. I will leave it for now though since there are already lots of tests that expect this data. But I will change it later.